File: Week_3/problem_8-Count_letters.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219856/problem/J

s = input()

v = [0] * 26  # Initialize a 26-sized list to store frequency of each letter

# using frequency array to count
for char in s:
    # Increment frequency of the character
    v[ord(char) - ord('a')] += 1  

    # The ord() function in Python is used to get the Unicode code point (integer value) of a given character.

    # ord(char) returns the Unicode code point of the character char.
    # ord('a') returns the Unicode code point of the letter 'a', which is 97.


for i in range(26):
    if v[i] != 0:
        # Print the character and its frequency
        print(f"{chr(i + ord('a'))} : {v[i]}")  

File: Week_3/problem_8-Count_letters.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/219856/problem/J

#include <bits/stdc++.h>
using namespace std;

int main()
{
    vector<int> v(26, 0); // Initialize an 26-sized array to store frequency of each letter

    string s;
    cin >> s;

    // using frequency array to count
    for (int i = 0; i < s.size(); i++)
    {
        v[s[i] - 'a']++; // Increment frequency of the character
    }

    for (int i = 0; i < 26; i++)
    {
        if (v[i] != 0)
        {
            // Print the character and its frequency
            cout << char(i + 'a') << " : " << v[i] << endl;
        }
    }

    return 0;
}
